reward/staking_manager.go

```diff
@@ -49,40 +51,68 @@ type StakingManager struct {
 	chainHeadSub         event.Subscription
 }
 
+var (
+	once           sync.Once
+	stakingManager *StakingManager
+)
+
+// NewStakingManager creates and returns StakingManager.
+//
+// On the first call, a StakingManager is created with given parameters.
+// From next calls, the existing StakingManager is returned. (Parameters
+// from the next calls will not affect.)
 func NewStakingManager(bc blockChain, gh governanceHelper, db stakingInfoDB) *StakingManager {
-	return &StakingManager{
-		addressBookConnector: newAddressBookConnector(bc, gh),
-		stakingInfoCache:     newStakingInfoCache(),
-		stakingInfoDB:        db,
-		governanceHelper:     gh,
-		blockchain:           bc,
-		chainHeadChan:        make(chan blockchain.ChainHeadEvent, chainHeadChanSize),
+	if bc != nil && gh != nil {
```

Comment:
Why the existence of `db` is not checked?

Reply:
StakingManager can work even if there is no DB. It is not mandatory to use it.
Moreover, `db` is set to `nil` in test file.

Reply:
If DB is not mandatory, you should adjust the log levels related to `addStakingInfoToDB`.
Without DB, Klaytn will print "WARN" level logs on every time `addStakingInfoToDB` called.

Reply:
Good point :) I will lower logging level to debug for `addStakingInfoToDB` and `getStakingInfoFromDB`.